### ISSUE TITLE
stats.End.Error does not implement status.Status

### ISSUE THREAD

**Issue by grantr** [grpc/grpc-go#1416]
Please answer these questions before submitting your issue.

### What version of gRPC are you using?

1.5.1

### What version of Go are you using (`go version`)?

 go1.8.1 darwin/amd64

### What operating system (Linux, Windows, …) and version?

MacOS 10.12.5

### What did you do?

The stats documentation at https://github.com/grpc/grpc-go/blob/b463cc3276f7d8a556664a893915d0f49d449a4d/stats/stats.go#L172 states:

> Error is the error just happened.  It implements status.Status if non-nil.

Since the type of `Error` is `error`, I tried casting it to `status.Status` like this:

```go
if sts, ok := event.Error.(status.Status); ok {
  // do stuff with sts
}
```

### What did you expect to see?

I expected to be able to cast the error to a status.Status and call `Code()` on it :smile:

### What did you see instead?

this error from the compiler:

```
impossible type assertion:
	".../google.golang.org/grpc/status".Status does not implement error (missing Error method)
```

**Comment by menghanl** [comment]
`It implements status.Status` is not accurate.
It should be something like `It is an error generated from status.Status and can be converted back to status.Status.`
[`status.FromError`](https://godoc.org/google.golang.org/grpc/status#FromError) should work for this error.
Sorry for the confusion. This should be fixed in #1418.

**Comment by grantr** [comment]
Thanks @menghani! This now compiles:

```go
if sts, ok := status.FromError(event.Error); ok {
  //do stuff with sts
}
```
